src/runtime/providers/cloudinary.ts

```diff
@@ -1,111 +1,126 @@
-      fill: 'fill',
-      inside: 'pad',
-      outside: 'lpad',
-      cover: 'fit',
-      contain: 'scale',
-      minCover: 'mfit',
-      minInside: 'mpad',
-      thumbnail: 'thumb',
-      cropping: 'crop',
-      coverLimit: 'limit'
-    },
-    format: {
-      jpeg: 'jpg'
-    },
-    background (value: string) {
-      return convertHexToRgbFormat(value)
-    },
-    color (value: string) {
-      return convertHexToRgbFormat(value)
-    },
-    gravity: {
-      auto: 'auto',
-      subject: 'auto:subject',
-      face: 'face',
-      sink: 'sink',
-      faceCenter: 'face:center',
-      multipleFaces: 'faces',
-      multipleFacesCenter: 'faces:center',
-      north: 'north',
-      northEast: 'north_east',
-      northWest: 'north_west',
-      west: 'west',
-      southWest: 'south_west',
-      south: 'south',
-      southEast: 'south_east',
-      east: 'east',
-      center: 'center'
-    }
-  },
-  joinWith: ',',
-  formatter: (key, value) => `${key}_${value}`
-})
+// const operationsGenerator = createOperationsGenerator({
+//   keyMap: {
+//     fit: 'c',
+//     width: 'w',
+//     height: 'h',
+//     format: 'f',
+//     quality: 'q',
+//     background: 'b',
+//     rotate: 'a',
+//     roundCorner: 'r',
+//     gravity: 'g',
+//     effect: 'e',
+//     color: 'co',
+//     flags: 'fl',
+//     dpr: 'dpr',
+//     opacity: 'o',
+//     overlay: 'l',
+//     underlay: 'u',
+//     transformation: 't',
+//     zoom: 'z',
+//     colorSpace: 'cs',
+//     customFunc: 'fn',
+//     density: 'dn',
+//     aspectRatio: 'ar'
+//   },
+//   valueMap: {
+//     fit: {
+//       fill: 'fill',
+//       inside: 'pad',
+//       outside: 'lpad',
+//       cover: 'fit',
+//       contain: 'scale',
+//       minCover: 'mfit',
+//       minInside: 'mpad',
+//       thumbnail: 'thumb',
+//       cropping: 'crop',
+//       coverLimit: 'limit'
+//     },
+//     format: {
+//       jpeg: 'jpg'
+//     },
+//     background (value: string) {
+//       return convertHexToRgbFormat(value)
+//     },
+//     color (value: string) {
+//       return convertHexToRgbFormat(value)
+//     },
+//     gravity: {
+//       auto: 'auto',
+//       subject: 'auto:subject',
+//       face: 'face',
+//       sink: 'sink',
+//       faceCenter: 'face:center',
+//       multipleFaces: 'faces',
+//       multipleFacesCenter: 'faces:center',
+//       north: 'north',
+//       northEast: 'north_east',
+//       northWest: 'north_west',
+//       west: 'west',
+//       southWest: 'south_west',
+//       south: 'south',
+//       southEast: 'south_east',
+//       east: 'east',
+//       center: 'center'
+//     }
+//   },
+//   joinWith: ',',
+//   formatter: (key, value) => `${key}_${value}`
+// })
 
 const defaultModifiers = {
   format: 'auto',
   quality: 'auto'
 }
 
 export const getImage: ProviderGetImage = (src, { modifiers = {}, baseURL = '/' } = {}) => {
+  const cloudName = baseURL.replace('https://res.cloudinary.com/', '').split('/')[0]
```

Comment:
one thought heer, and this could be considered out of scope if it doesnt currently support it, is that someone could be potentially passing in a URL that's not from `res.cloudinary.com` such as if they're using they're own cname

https://cloudinary.com/documentation/advanced_url_delivery_options#private_cdns_and_cnames

i can't remember if my loader technically supports that, i can't remember, but if it does, you could potentially use the parUrl option in the adjacent library (dep of url-loader) @cloudinary-util/util which can grab all the parts, including the host

ill need to add test cases for that to ensure its working properly

Reply:
I would say that this is a more advanced scenario that could be added once requested by the community :) At least IMO.